agent/ecr/client.go

```diff
@@ -67,3 +76,21 @@ func (client *ecrClient) GetAuthorizationToken(registryId string) (*ecrapi.Autho
 
 	return authData, nil
 }
+
+// Ensure token is still within it's expiration window. We early expire to allow for timing in calls and add jitter to avoid
+// refreshing all of the tokens at once.
+func (client *ecrClient) isTokenValid(authData *ecrapi.AuthorizationData) bool {
+	return authData != nil &&
+		authData.ExpiresAt != nil &&
+		aws.TimeValue(authData.ExpiresAt).Add(-1*client.expirationWindowWithJitter()).After(time.Now())
+}
+
+const (
```

Comment:
Sounds good - wasn't sure what we prefer for style on agent (const close to use or const at top of file)